src/harness/harness.ts

```diff
@@ -1557,21 +1557,32 @@ module Harness {
         export interface BaselineOptions {
             LineEndingSensitive?: boolean;
             Subfolder?: string;
+            Baselinefolder?: string;
         }
 
-        export function localPath(fileName: string, subfolder?: string) {
-            return baselinePath(fileName, 'local', subfolder);
+        export function localPath(fileName: string, baselinefolder?: string, subfolder?: string) {
+            if (baselinefolder === undefined) {
+                return baselinePath(fileName, 'local', 'tests/baselines', subfolder);
+            }
+            else {
+                return baselinePath(fileName, 'local', baselinefolder, subfolder);
+            }
         }
 
-        function referencePath(fileName: string, subfolder?: string) {
-            return baselinePath(fileName, 'reference', subfolder);
+        function referencePath(fileName: string, baselinefolder?: string, subfolder?: string) {
+            if (baselinefolder === undefined) {
+                return baselinePath(fileName, 'reference', 'tests/baselines', subfolder);
+            }
+            else {
+                return baselinePath(fileName, 'reference', baselinefolder, subfolder);
+            }
         }
 
-        function baselinePath(fileName: string, type: string, subfolder?: string) {
+        function baselinePath(fileName: string, type: string, baselinefolder: string, subfolder?: string) {
             if (subfolder !== undefined) {
-                return Harness.userSpecifiedroot + 'tests/baselines/' + subfolder + '/' + type + '/' + fileName;
+                return Harness.userSpecifiedroot + baselinefolder + '/' +  subfolder + '/' + type + '/' + fileName;
             } else {
-                return Harness.userSpecifiedroot + 'tests/baselines/' + type + '/' + fileName;
+                return Harness.userSpecifiedroot + baselinefolder + '/'  + type + '/' + fileName;

```

Comment:
I just realized how much of a headache this is going to be. Use `path.join`.

Really we should be using `path.join` instead of any string concatenation when possible.